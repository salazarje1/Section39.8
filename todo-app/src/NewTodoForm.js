import React, { useState } from 'react'; 

const NewItemForm = ({ addTask }) => {
    const [task, setTask] = useState(''); 

    const handleChange = (e) => {
        setTask(e.target.value); 
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        addTask(task); 
        setTask(''); 
    }

    return (
        <form onSubmit={handleSubmit}>
            <label>
                New Task
                <input
                    type="text"
                    name="todo"
                    placeholder='todo'
                    value={task}
                    onChange={handleChange}
                />
            </label>
            <button>Add New Task</button>
        </form>
    )
}

export default NewItemForm; 