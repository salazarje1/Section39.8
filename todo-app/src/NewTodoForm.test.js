import { render, fireEvent } from "@testing-library/react";
import NewTodoForm from './NewTodoForm';

it('should render without crashing', () => {
    render(<NewTodoForm />); 
})

it('should match with snapshot', () => {
    const { asFragment } = render(<NewTodoForm />);
    expect(asFragment()).toMatchSnapshot(); 
})