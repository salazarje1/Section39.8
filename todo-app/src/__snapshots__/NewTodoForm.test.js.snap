// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match with snapshot 1`] = `
<DocumentFragment>
  <form>
    <label>
      New Task
      <input
        name="todo"
        placeholder="todo"
        type="text"
        value=""
      />
    </label>
    <button>
      Add New Task
    </button>
  </form>
</DocumentFragment>
`;
